src/data/finset/basic.lean

```diff
@@ -1447,6 +1447,13 @@ end
 lemma filter_nonempty_iff {s : finset α} : (s.filter p).nonempty ↔ ∃ a ∈ s, p a :=
 by simp only [nonempty_iff_ne_empty, ne.def, filter_eq_empty_iff, not_not, not_forall]
 
+lemma filter_eq_self (s : finset α) :
+  s.filter p = s ↔ ∀ x ∈ s, p x :=
```

Comment:
Do you want to move `filter_true_of_mem` below this and use this to prove that? Also, I'll trust your word, but just to make sure, is it easier to prove this with `rw finset.ext_iff` and some logic lemmas?

Reply:
Thanks! these suggestions worked well